runtime/storage-proxy.js

```diff
@@ -255,43 +262,76 @@ class CollectionProxy extends StorageProxyBase {
   }
 }
 
+// Variables are synchronized in a 'last-writer-wins' scheme. When the
+// VariableProxy mutates the model, it sets a barrier and expects to 
+// receive the barrier value echoed back in a subsequent update event.
+// Between those two points in time updates are not applied or
+// notified about as these reflect concurrent writes that did not 'win'.
 class VariableProxy extends StorageProxyBase {
   constructor(...args) {
     super(...args);
     this._model = null;
+    this._barrier = null;
   }
   _synchronizeModel(model) {
     assert('data' in model);
     this._model = model.data;
   }
   _updateModel(update) {
     assert('data' in update);
+    // If we have set a barrier, suppress updates until after
+    // we have seen the barrier return via an update.
+    if (this._barrier && update.barrier != this._barrier) {
```

Comment:
Can 0 be a valid barrier value? Probably should have an explicit check against null here.

Reply:
Done.